### PR TITLE
Release Traefik 2.3

### PR DESCRIPTION
**Pull request by SantoDE**
This PR updates Trafik to 2.3, including the required changes for the new IngressClass feature.

Superseeds: #246 / #256

### CODE REVIEW

**Comment by SantoDE** [comment]
Ping @DirtyCajunRice @myoung34 - I merged the PR's together to this one. PTAL if you have time :)

File: traefik/templates/rbac/clusterrole.yaml [hunk 1]
```diff
@@ -31,6 +33,7 @@ rules:
       - extensions
     resources:
       - ingresses/status
+      - networking.k8s.io
```
Comment (DirtyCajunRice) [inline traefik/templates/rbac/clusterrole.yaml:36]: This is an ApiGroup not a resource and needs to be removed

File: traefik/templates/ingressclass.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    app.kubernetes.io/name: {{ template "traefik.name" . }}
+    helm.sh/chart: {{ template "traefik.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  name: {{ template "traefik.fullname" . }}
+spec:
+  controller: traefik.io/ingress-controller
+    {{- end }}
+    {{- end }}
```
Comment (DirtyCajunRice) [inline traefik/templates/ingressclass.yaml:17]: newline at the end of the file

**Comment by SantoDE** [comment]
Maybe @notsureifkevin as well :-)

**Review by DirtyCajunRice — CHANGES_REQUESTED** [review]
Other than these requested changes, if you want this to be == in feature to the 2.3 release you should add a top level 
```yaml
pilot:
  token: “”
```
and an if statement enabling it as an argument. Other than that it looks good!

File: traefik/templates/ingressclass.yaml (hunk 2 above)
Comment (SantoDE) [reply traefik/templates/ingressclass.yaml]: Done

File: traefik/templates/rbac/clusterrole.yaml (hunk 1 above)
Comment (SantoDE) [reply traefik/templates/rbac/clusterrole.yaml]: Done

**Comment by SantoDE** [comment]
Hmm, good catch @DirtyCajunRice. I think we should do it :)

**Comment by SantoDE** [comment]
Done @DirtyCajunRice

File: traefik/templates/rbac/clusterrole.yaml [hunk 3]
```diff
@@ -21,14 +21,17 @@ rules:
       - watch
   - apiGroups:
       - extensions
+      - networking.k8s.io
     resources:
       - ingresses
+      - ingressclasses
```
Comment (notsureifkevin) [inline traefik/templates/rbac/clusterrole.yaml:27]: would this break if defined on a cluster that is not `v1.18`?

Comment (SantoDE) [reply traefik/templates/rbac/clusterrole.yaml:27]: I think it would just not work :)

**Review by ldez — APPROVED** [review]
LGTM

**Comment by tobylo** [comment]
What and where is the name of the controller specified? How would a setup look like when running multiple instances of traefik with the IngressClass resource?

**Pull request merged**